modules/xcm-bridge-hub-router/src/lib.rs

```diff
@@ -109,29 +113,32 @@ pub mod pallet {
 	#[pallet::hooks]
 	impl<T: Config<I>, I: 'static> Hooks<BlockNumberFor<T>> for Pallet<T, I> {
 		fn on_initialize(_n: BlockNumberFor<T>) -> Weight {
-			// if XCM channel is still congested, we don't change anything
-			if T::LocalXcmChannelManager::is_congested(&T::SiblingBridgeHubLocation::get()) {
-				return T::WeightInfo::on_initialize_when_congested()
-			}
-
-			// if we can't decrease the delivery fee factor anymore, we don't change anything
-			let mut delivery_fee_factor = Self::delivery_fee_factor();
-			if delivery_fee_factor == MINIMAL_DELIVERY_FEE_FACTOR {
-				return T::WeightInfo::on_initialize_when_congested()
-			}
-
-			let previous_factor = delivery_fee_factor;
-			delivery_fee_factor =
-				MINIMAL_DELIVERY_FEE_FACTOR.max(delivery_fee_factor / EXPONENTIAL_FEE_BASE);
-			log::info!(
-				target: LOG_TARGET,
-				"Bridge channel is uncongested. Decreased fee factor from {} to {}",
-				previous_factor,
-				delivery_fee_factor,
-			);
-
-			DeliveryFeeFactor::<T, I>::put(delivery_fee_factor);
-			T::WeightInfo::on_initialize_when_non_congested()
+			/*
```

Comment:
Now we have multiple delivery fee factors - one per every bridge. So the most straightforward solution to decrease this fee factor from `on_initialize` is to loop over all bridges in every block `on_initialize` and decrease it. However it seems too dumb. Especially, given that `is_congested` call will return the same value for all bridges.

So what I'm currently thinking of is to have two fee factors. The first one will be shared by all bridges and will be decreased from `on_initialize` only when the channel with bridge hub is uncongested - let's call it `CongestionFeeFactor`. The other one will be per-bridge. We'll be decreasing it when a message is sent over the bridge. It'll be the `BridgeFeeFactor`.

So the total fee would be the `(base_fee + msg_len * byte_fee) * (CongestionFeeFactor + BridgeFeeFactor - 1)`.